### PR TITLE
dbotconf: Fix Windows produces backslashes instead of slashes

### LINKED ISSUE #264: dbotconf: on Windows it produces backslashes instead of slashes
Using `dbotconf` on Windows results in having backslashes in `directory` field instead of slashes

E.g. 

```
  - package-ecosystem: docker
    directory: \example\zipkin
```

instead of

```
  - package-ecosystem: docker
    directory: /example/zipkin
```

I guess the code misses [`filepath.ToSlash()`](https://pkg.go.dev/path/filepath#ToSlash) in some place(s).

### PR DESCRIPTION
**Pull request by scorpionknifes**
Fixes #264

@pellared 
> `multimod tag` is also not working correctly because of backslashes on Windows.
> 
> I guess `crosslink` will have the same issue.

Shall look into `multimod` and `crosslink`

### CODE REVIEW

**Review by pellared — COMMENTED** [review]
First of all thanks for the PR 👍 

1. I see that there are some build failures.
2. What do you think about adding a GitHub Workflow (or job in existing workflow - probably this would be better) that would execute tests on Windows?

File: CHANGELOG.md
Comment (pellared) [reply CHANGELOG.md:6]: I created https://github.com/open-telemetry/opentelemetry-go-build-tools/issues/449

File: .gitignore
Comment (pellared) [inline .gitignore:17]: You can change it to a single rule:

```
*.exe
```

File: internal/syncerror/known_sync_error_windows.go
Comment (scorpionknifes) [inline internal/syncerror/known_sync_error_windows.go:31]:
```
internal\syncerror\known_sync_error_windows.go:27:9: comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error (errorlint)
        return err == windows.ERROR_INVALID_HANDLE
```
Interesting, didn't know golangci-lint doesn't check go files for Windows when ran on unix systems (ci).

**Comment by pellared** [comment]
> Shall look into multimod and crosslink.

Can you quickly test it and create issues if these tools have the same bug?

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
Thanks a lot for your PR 👍 Awaiting second review.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
